examples/tensorflow-unit-testing/README.md

```diff
@@ -0,0 +1,142 @@
+needs more debugging. It’s typically a good idea to add a unit test each time you fix any specific
+error to make sure this error wouldn’t occur again.
+
+## Testing custom layers
+When you implement custom training routines, the recommended
+practice is to create
+[custom layers](https://www.tensorflow.org/guide/keras/custom_layers_and_models)
+by subclassing [tf.keras.layers.Layer](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer).
+This gives you a possibility to isolate (and test) logical pieces of your code, which makes it more
+readable and easier to maintain.
+
+Let's play with `ExampleBlock` - a simple custom layer. First, we’d like to test whether the shape of the output tensor is the one we expect.
+```python
+class LinearBlockTest(tf.test.TestCase):
+   def test_shape_default(self):
+       x = np.ones((4, 32))
+       layer = example.LinearBlock()
+       output = layer(x)
+       self.assertAllEqual(output.shape, (4, 32))
+```
+
+You can find more examples testing output shape by exploring the `LinearBlockTest` class.
+
+The next thing we can also check is the actual output. It’s not always needed but it might be a good
+idea when you have a layer with a custom logic that needs to be double-checked.
+Please note that despite using initializers for weights, our test is not flaky (tf.test.TestCase
+takes care about it).
+We can also patch various pieces of the layer we’re concerned about (other layers used, loss
+functions, stdout, etc.) to check the desired output. Let’s have a look at the example where we’ve
+patched initializer:
+```python
+@patch.object(initializers, 'get', lambda _: tf.compat.v1.keras.initializers.Ones)
+def test_output_ones(self):
+   dim = 4
+   batch_size = 3
+   output_dim = 2
+   x = np.ones((batch_size, dim))
+   layer = example.LinearBlock(output_dim)
+   output = layer(x)
+   expected_output = np.ones((batch_size, output_dim)) * (dim + 1)
+   self.assertAllClose(output, expected_output, atol=1e-4)
+```
+
+## Testing custom keras models
+The easiest way to test your model is to prepare a small fake dataset and run a few training steps on
+this dataset to check whether the model can be successfully trained, and validation and prediction
+also works. Please keep in mind, that successfully means all steps can be run without generating
+errors, but in the basic case we don’t check whether the training itself makes sense - i.e., whether
+a loss decreases to any meaningful value. But more about it later.
+Let’s have a look at a simple example of how to test a model from this [tutorial](https://www.tensorflow.org/tutorials/keras/regression).
+```python
+class ExampleModelTest(tf.test.TestCase):
+   def _get_data(self):
+       dataset_path = tf.keras.utils.get_file(
+           'auto-mpg.data',
+           'http://archive.ics.uci.edu/ml/machine-learning-databases/auto-mpg/auto-mpg.data')
+       column_names = ['MPG','Cylinders','Displacement','Horsepower','Weight',
```

Comment:
fyi this doesn't match the code after the style fixes were applied

Reply:
Thanks, let's leave as it is for now!